### ISSUE TITLE
Using a game engine

### ISSUE THREAD

**Issue by abmyii** [remance/Masendor#1]
Hello! I saw your game on the Pygame website and it caught my eye - the concept is great and I'm impressed by your progress already! I've made my fair share of games in Pygame and I'm aware of how bare-bone it is - as you may know, it isn't really a game engine. Due to a lack of spare time I've moved on from dabbling in gamedev, but since, I've come to realise that using a "proper" game engine is almost critical if you want your game to scale - writing your own game engine (which you practically must do when using Pygame) is really not a good idea, unless you are really experienced.

I'm not trying to put you off, but in my humble opinion I think using a game engine will get your game a lot further, far quicker, especially when it comes tasks like making the terrain, etc. (your note about it being difficult and long is very accurate!). My personal choice would be Godot for many reasons, amongst others that it is open source, has a very active and productive community (https://www.reddit.com/r/godot/top/?t=all) and has most of the features of professional game engines. It's native scripting language is not Python, but I believe there are addons which allow you to script it using Python. There are also plugins, demos and tutorials for (procedural) terrain generation. Some examples of plugins are https://github.com/Zylann/godot_heightmap_plugin and https://github.com/SirRamEsq/SmartShape2D)

Whatever you decide (and I'm aware the decision will not be a simple one - one that even I would find difficult due to how much I enjoy working with Pygame despite the challenges), I wish you all the best with your project, and I can't wait to see how it turns out!

**Comment by remance** [comment]
Hello and thank you for the suggestion on Godot engine, I will keep that into consideration for sure. I will definitely encounter performance issues when the game is getting bigger especially in ver 5.0. For now, I will see how far pygame can get me without having to cut too much corner.

**Comment by abmyii** [comment]
Came across this - a Godot game with a very similar concept to this!
https://kaetjaatyy.github.io/2020/10/27/battles.html

**Comment by remance** [comment]
Indeed, seem like we both have the same idea especially in term of unit structure. Too bad there is not video showcase yet but seem like the way battle progress is probably also in the same way. Definitely worth following.

**Comment by legotrainkid** [comment]
If you want to continue using python, try arcade. I haven't personally used it, but I am thinking about using it instead of pygame for one of my games. Here's the link: https://arcade.academy/

**Comment by remance** [comment]
Look interesting. Wonder how it fares against Pygame 2.0 now though. But the library does look somewhat simpler to use than Pygame. In any case, thanks for the suggestion never know about it before.
